### DISCUSSION TITLE
compiling for ESP32C3 whith IRremote

### DISCUSSION THREAD

**Discussion by akluchnikov** [MoonModules/WLED-MM#202] [Q&A] [answered]
Hello everyone. I want to add infrared remote control functions to the firmware. I've been struggling for a week, but I can't compile from PIO. Everything ends up in error
```

Resolving esp32c3dev_4MB_M dependencies...
Library Manager: Installing git+https://github.com/Aircookie/ESPAsyncWebServer.git @ 2.2.1
git version 2.47.1.windows.1
Cloning into 'C:\.platformio\.cache\tmp\pkg-installing-ce_kf9y7'...

PackageException: Package version 2.3.0+sha.00be156 doesn't satisfy requirements 2.2.1 based on PackageMetadata <type=library name=ESPAsyncWebServerWLED version=2.3.0+sha.00be156 spec={'owner': None, 'id': None, 'name': 'ESPAsyncWebServer', 'requirements': '2.2.1', 'uri': 'git+https://github.com/Aircoookie/ESPAsyncWebServer.git'}
```

I tried versions b28, b30, b32, tried to substitute links to libraries in different ways and even download them myself, as well as compile via Arduino IDE. Tell me what to do, please

**Comment by dosipod** [comment]
@akluchnikov 
look at platformio.ini here 
https://github.com/MoonModules/WLED/blob/mdev/platformio.ini
Then change ESPAsyncWebServer reference to 
https://github.com/Aircoookie/ESPAsyncWebServer.git#v2.2.1
![image](https://github.com/user-attachments/assets/927670eb-823c-4654-b4e2-ac9355779d30)

**Reply by akluchnikov** [reply]
Thank you very much for the answer!
But it didn't help. And the error code looks more like a mockery )))
![image](https://github.com/user-attachments/assets/4503fbb6-1178-413b-acb0-37de581cbfcf)
Maybe there is a way to manually insert the library?
The project has a complex build system, I have never encountered such a thing, I apologize for my stupid questions and mistakes

**Reply by akluchnikov** [reply]
I thought I misunderstood you, here's another try. The error is the same with versions 2.2.1 and 2.3.0. Although these files are present in the repository https://github.com/Aircoookie/ESPAsyncWebServer.git
![image](https://github.com/user-attachments/assets/78452a3e-5115-49c4-9bc5-a700643d36f0)

**Reply by dosipod** [reply, marked as answer]
Yes because you did not write it the same as it is in the  ini file linked to you , add the below exactly 
https://github.com/Aircoookie/ESPAsyncWebServer.git#v2.2.1

**Reply by akluchnikov** [reply]
Thanks for your help!
I took the wled-mdev version from github, it compiled and even loaded into the device. However, it does not work. I suppose something is wrong with the bootloader or memory layout. Before flashing, I deleted everything using the esp flash download tool.
![image](https://github.com/user-attachments/assets/ab39704a-fc32-4fef-ac9b-6dff8b653856)

I also tried to download the bootloader, the layout file and the firmware via the esp flash download tool. The result is the same.
![image](https://github.com/user-attachments/assets/0e415fc6-aa8e-4ddf-872c-c799f78df7c9)

I tried to install the firmware from the web installer install.wled.me, it worked (but for some reason the leds do not work, although the port is selected correctly). I tried to download firmware.bin via the web interface, but the system said that the file is incorrect and refused to update.

Please tell me what else I can do to make this work.

**Reply by dosipod** [reply]
@akluchnikov  You can install from this site https://wled-install.github.io/  or using this tool https://github.com/srg74/WLED-wemos-shield/tree/master/resources/Firmware/WLED_%20ESP_Flasher  .

**Reply by akluchnikov** [reply]
Thanks for the advice!
If I install this version via wled-install.github.io it works.

![image](https://github.com/user-attachments/assets/e3c0fbc9-e4c4-40fe-ba60-4a0425333614)

If I install the MM version it doesn't work

![image](https://github.com/user-attachments/assets/cc47761d-8583-40a6-8443-bacee7bba1a4)

There are endless messages in the serial port like:
```
rst:0x3 (RTC_SW_SYS_RST),boot:0xf (SPI_FAST_FLASH_BOOT)
Saved PC:0x403d10fc
SPIWP:0xee
mode:DOUT, clock div:1
load:0x3fcd6100,len:0x38c
load:0x403ce000,len:0x6ac
load:0x403d0000,len:0x24e4
entry 0x403ce000

```
if I download the firmware, and not install it and install it via ESPflasher, the first one works, but the second one also does not work. Updating via OTA from the web interface also does not work - the system says that the file is incorrect

**Reply by dosipod** [reply]
I forgot to mention that we faced the same issue when trying to add audio to C3 as the bin is too big , Since MM has audio already ,issue opened here  https://github.com/MoonModules/WLED/issues/173 . 

For AC if you need audio then you have to drop other things same as the below example 
```ini
[env:esp32c3dev_AR]
extends = esp32c3
platform = ${esp32c3.platform}
platform_packages = ${esp32c3.platform_packages}
framework = arduino
board = esp32-c3-devkitm-1
board_build.partitions = ${esp32.default_partitions}
build_flags = ${common.build_flags} ${esp32c3.build_flags} ${esp32.AR_build_flags} -D WLED_RELEASE_NAME=esp32c3dev_AR
  -D WLED_WATCHDOG_TIMEOUT=0
  -DLOLIN_WIFI_FIX ; seems to work much better with this
  -DARDUINO_USB_CDC_ON_BOOT=1 ;; for virtual CDC USB
  ;-DARDUINO_USB_CDC_ON_BOOT=0   ;; for serial-to-USB chip
  -D WLED_DISABLE_ALEXA 
  -D WLED_DISABLE_MQTT  
  -D WLED_DISABLE_ADALIGHT 
  -D WLED_DISABLE_LOXONE 
  -D WLED_DISABLE_HUESYNC
upload_speed = 460800
build_unflags = ${common.build_unflags}
lib_deps = ${env:esp32c3dev.lib_deps}  
  ${esp32.AR_lib_deps}
```

**Reply by akluchnikov** [reply]
Thanks everyone for your help!

I deleted vscode, deleted c:\.pltatformio , reinstalled everything and with this config everything worked

```ini
[env:seeed_esp32c3_4MB_S]
extends = env:esp32c3dev_4MB_M
board = seeed_xiao_esp32c3
platform = ${esp32.platformV4_pre}                             ;; standard IDF 4.4.1 platform - seems to work better for seeed xiao board
platform_packages = ${esp32.platformV4_packages_pre}
board_build.flash_mode = qio
upload_speed = 460800
build_unflags = ${env:esp32c3dev_4MB_M.build_unflags}
  -DWLEDMM_FASTPATH ;; needs more testing on -C3
  -D WLED_ENABLE_HUB75MATRIX ;; not enough pins
build_flags = ${common.build_flags} ${esp32c3.build_flags} 
  -D WLED_WATCHDOG_TIMEOUT=0 -D CONFIG_ASYNC_TCP_USE_WDT=0
  ${common_mm.build_flags_S} -Wno-misleading-indentation -Wno-format-truncation
  -D WLED_RELEASE_NAME=seeed_esp32c3_4MB_S
  -DARDUINO_USB_CDC_ON_BOOT=1  ;; enable CDC USB -> needed for debugging over serial USB
  -D WLED_DISABLE_ADALIGHT     ;; to disable serial protocols when using CDC USB (Serial RX pin will receive junk commands, unless its pulled down by resistor)
  ;-D WLED_DISABLE_INFRARED  ;; save flash space
  -D WLED_DISABLE_ALEXA    ;; save flash space
  -D WLED_DISABLE_HUESYNC  ;; save flash space
  -D WLED_DISABLE_LOXONE   ;; save flash space
  ;-D WLEDMM_WIFI_POWERON_HACK -DLOLIN_WIFI_FIX ;; use this _only_ if your device is not able to make a WiFI connection!
  -D LEDPIN=3  ;; attach your LEDs to GPIO3 aka "D1" / "A1"
  -D BTNPIN=4
  -D RLYPIN=-1 -D IRPIN=2
  -D SR_DMTYPE=1 
  -D I2S_SDPIN=5 -D I2S_WSPIN=6 -D I2S_CKPIN=8 -D MCLK_PIN=7
  -D WLED_USE_MY_CONFIG
  ;-D WLED_DEBUG -D SR_DEBUG
lib_deps = ${esp32c3.lib_deps} ${common_mm.lib_deps_S}
;lib_ignore = IRremoteESP8266 ; use with WLED_DISABLE_INFRARED for faster compilation
monitor_filters = esp32_exception_decoder
```
